internal/pkg/daemon/bpfrecorder/bpfrecorder.go

```diff
@@ -447,19 +480,50 @@ func (b *BpfRecorder) processEvents(events chan []byte) {
 		}
 
 		if profile, exist := b.profileForContainerIDs.LoadAndDelete(containerID); exist {
-			rawComm, err := b.GetValue(b.comms, pid)
-			if err != nil {
-				b.logger.Error(err, "unable to get command name for PID", "pid", pid)
+			b.logger.Info(
+				"Saving PID for profile",
+				"pid", pid, "mntns", mntns, "profile", profile,
+			)
+			b.trackProfileForPid(pid, mntns, profile)
+			if mntns != b.systemMountNamespace {
+				b.profileForMountNamespace.Store(mntns, profile)
 			}
-			comm := toStringByte(rawComm)
-			b.logger.Info("Saving PID for profile", "pid", pid, "profile", profile, "comm", comm)
-
-			pids, _ := b.pidsForProfiles.LoadOrStore(profile, []Pid{})
-			b.pidsForProfiles.Store(profile, append(pids.([]Pid), Pid{id: pid, comm: comm}))
 		}
 	}
 }
 
+func (b *BpfRecorder) findSystemMountNamespace() (uint64, error) {
+	res, err := b.Readlink("/proc/1/ns/mnt")
```

Comment:
May be worth adding a comment highlighting that this requires `hostPID`.

Reply:
Following-up in https://github.com/kubernetes-sigs/security-profiles-operator/pull/647